core/controllers/topic_editor.py

```diff
@@ -120,12 +120,13 @@ def get(self, topic_id, subtopic_id):
             topic_id, subtopic_id, strict=False)
 
         if subtopic_page is None:
-            raise self.PageNotFoundException(
-                Exception('The subtopic with the given id doesn\'t exist.'))
-
-        self.values.update({
-            'subtopic_page': subtopic_page.to_dict()
-        })
+            self.values.update({
+                'subtopic_page': None
```

Comment:
er, why not 404?

Reply:
In the frontend, when Create Subtopic is clicked, it creates a local subtopic object, so, when its subtopic page is requested, it would have given a 404. So, I am now returning None, and based on that creating a local subtopic page object as well.
The subtopic page is actually created in the backend only after updating the topic, when the 'create_subtopic' is encountered.

Reply:
But if the subtopic object is created locally, then there should be no request to the backend at all to get the subtopic. The frontend should already know which is which -- right?

Reply:
The only indication that a subtopic is newly created for the frontend is its presence in the changelist. How about I push the new subtopic page into _subtopicPages when its created? But then here, the page has to be deleted also when the newly created subtopics are deleted. Shall I do this?

Reply:
You should have some kind of frontend cache that is initialized with all the subtopic indices that are  currently part of the topic, and that gets updated when subtopics are created/deleted in the frontend. Then use that cache as the source of truth.

If the above describes _subtopicPages then it is fine to use _subtopicPages as the source of truth. But if _subtopicPages doesn't contain e.g. pages whose content hasn't been fetched from the backend yet, then you'll need to find another way.

Reply:
That cache would be topic.subtopics itself, since it would have all the changes done locally, but for subtopic pages, initially, all the pages are not required, right? They are fetched as they are loaded. I'll push the new subtopic page to _subtopicPages which is the local cache for subtopic pages.

Reply:
Ah ok, that sounds fine then. So if I understand correctly, the source of truth for "what subtopics exist" is topic.subtopics, and the source of truth for "what is the content of subtopic page X, given that it exists" is _subtopicPages, right? That sounds fine to me.

Reply:
Yup!

Reply:
Done, added local create and delete subtopic pages. Also, did the decrement of subtopic ids of subsequent newly created subtopic pages when a newly created subtopic is deleted.